File: hw/carvajal/u2/hw19Inheritance/PythonFarmSystem/model/cow.py
from model.farm_animal import FarmAnimal

class Cow(FarmAnimal):
    def __init__(self, is_producing_milk: bool, liters_a_day: float, animal_id: int, breed: str,
                 born_on, gender: str, is_able_to_reproduce: bool,
                 weight: float, cage):
        super().__init__(animal_id, breed, born_on, gender,
                         is_able_to_reproduce, weight, cage)
        self.is_producing_milk = is_producing_milk
        self.liters_a_day = liters_a_day

    def milk(self):
        return self.liters_a_day if self.is_producing_milk else 0

    def __str__(self):
        return f"Cow(producing={self.is_producing_milk}, liters={self.liters_a_day}, {super().__str__()})"

File: hw/vargas/u2/hw19Inheritance/java/Farmsystem/src/main/java/ec/edu/espe/farmsystem/model/Cow.java
package ec.edu.espe.farmsystem.model;

import java.util.Date;

/**
 *
 * @author César Vargas, Paradigm, @ESPE
 */
public class Cow extends FarmAnimal{
    private boolean isProducingMilk;
    private float littersADay;

    public Cow(boolean isProducingMilk, float littersADay, int id, String breed, Date bornOn, String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.isProducingMilk = isProducingMilk;
        this.littersADay = littersADay;
    }

    @Override
    public String toString() {
        return "Cow{" + "isProducingMilk=" + isProducingMilk + ", littersADay=" + littersADay + super.toString() + '}';
    }
    
    
    public float milk(){
        if(isIsProducingMilk()){
            setLittersADay(getLittersADay() + 1);

        }
        return getLittersADay();
    }

    /**
     * @return the isProducingMilk
     */
    public boolean isIsProducingMilk() {
        return isProducingMilk;
    }

    /**
     * @param isProducingMilk the isProducingMilk to set
     */
    public void setIsProducingMilk(boolean isProducingMilk) {
        this.isProducingMilk = isProducingMilk;
    }

    /**
     * @return the littersADay
     */
    public float getLittersADay() {
        return littersADay;
    }

    /**
     * @param littersADay the littersADay to set
     */
    public void setLittersADay(float littersADay) {
        this.littersADay = littersADay;
    }
    
    
}
